types/mongoose/test/model.ts

```diff
@@ -227,18 +227,24 @@ MongoModel.update({ age: { $gt: 18 } }, { oldEnough: true }, cb);
 MongoModel.update({ name: 'Tobi' }, { ferret: true }, { multi: true,  arrayFilters: [{ element: { $gte: 100 } }] }, cb);
 MongoModel.where('age').gte(21).lte(65).exec(cb);
 MongoModel.where('age').gte(21).lte(65).where('name', /^b/i);
-new (mongoModel.base.model(''))();
-mongoModel.baseModelName && mongoModel.baseModelName.toLowerCase();
+new (mongoModel.constructor.base.model(''))();
+// $ExpectError
+mongoModel.baseModelName;
+mongoModel.constructor.baseModelName && mongoModel.constructor.baseModelName.toLowerCase();
 mongoModel.collection.$format(99);
 mongoModel.collection.initializeOrderedBulkOp;
 mongoModel.collection.findOne;
 mongoModel.db.openUri('');
+// $ExpectError
 mongoModel.discriminators;
-mongoModel.modelName.toLowerCase();
-MongoModel = mongoModel.base.model('new', mongoModel.schema);
-/* inherited properties */
-MongoModel.modelName;
```

Comment:
I'm not sure if this, along with `base` should be removed. In your example:
```ts
const Foo = mongoose.model('Foo', new mongoose.Schema({ fooProp: String }));
Foo.modelName // 'Foo'
Foo.base // Mongoose
```

These attributes should still be valid and the tests should probably remain.

Reply:
It's tested via the consctructor property a few lines below, because `foo.constructor === Foo` statically with this PR.
if you think it's still worth it, I can duplicate it for all 4 properties `base` `baseModelName` `discriminators` `modelName`.

Reply:
Ah, I missed that! Makes sense to me!